### PR TITLE
Feature/add bll layer

### PR DESCRIPTION
**Pull request by Mmishaaa**
* Add SubscriptionService
* Add mapster for mapping
* Update the Service Repository a bit
* Add DateTimeProvider

### CODE REVIEW

File: SubscriptionService/SubscriptionService.BLL/Services/SubscriptionService.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            {
+                throw new BadRequestException("User already has subscription");
+            }
+
+            var modelToCreate = new SubscriptionEntity
+            {
+                FusionUserId = fusionUserId,
+                SubscriptionType = subscription.SubscriptionType,
+                CreatedAt = DateTime.Now,
+                UpdatedAt = DateTime.Now,
```
Comment (moltenessence) [inline SubscriptionService/SubscriptionService.BLL/Services/SubscriptionService.cs]: create _dateTime provider

File: SubscriptionService/SubscriptionService.BLL/Services/SubscriptionService.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+            var fusionUserId = Guid.Parse(userJson["id"].ToString());
+
+            var subscription = new Subscription
+            {
+                FusionUserId = fusionUserId,
+                SubscriptionType = SubscriptionType.Base,
+                CreatedAt = DateTime.Now,
+                UpdatedAt = DateTime.Now,
+                ExpiresAt = DateTime.Now.AddMonths(1)
```
Comment (moltenessence) [inline SubscriptionService/SubscriptionService.BLL/Services/SubscriptionService.cs]: same

File: SubscriptionService/SubscriptionService.BLL/Services/SubscriptionService.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            var entities = await _subscriptionRepository.GetAllAsync(cancellationToken);
+            return entities.Adapt<List<Subscription>>();
+        }
+
+        public async Task<Subscription> GetByIdAsync(Guid id, CancellationToken cancellationToken)
+        {
+            var entity = await _subscriptionRepository.GetByIdAsync(id, cancellationToken) ?? throw new NotFoundException("Entity with this id doesn't exist");
+            return entity.Adapt<Subscription>();
+        }
+        public async Task<Subscription> UpdateAsync(Guid id, SubscriptionType subscriptionType, CancellationToken cancellationToken)
```
Comment (moltenessence) [inline SubscriptionService/SubscriptionService.BLL/Services/SubscriptionService.cs]: formatting

File: SubscriptionService/SubscriptionService.BLL/Services/SubscriptionService.cs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        {
+            var subscriptionToUpdate = await _subscriptionRepository.GetByIdAsync(id, cancellationToken) ?? throw new NotFoundException("Entity with this id doesn't exist");
+            subscriptionToUpdate.SubscriptionType = subscriptionType;
+            subscriptionToUpdate.ExpiresAt = DateTime.Now.AddMonths(1);
+            subscriptionToUpdate.UpdatedAt = DateTime.Now;
```
Comment (moltenessence) [inline SubscriptionService/SubscriptionService.BLL/Services/SubscriptionService.cs]: provider

**Review by moltenessence — CHANGES_REQUESTED** [review]

**Review by moltenessence — APPROVED** [review]

**Review by DarPers — APPROVED** [review]

**Review by VadimZharikov — APPROVED** [review]

**Pull request merged**